gnosis-imap/gnosis-imap-spec.ini

```diff
@@ -1147,13 +1174,49 @@ proxy_origstorage: store(M2, #hashedLocation({COMPILER}, {NONCE}, .IntList),
     andBool SIGS_LEN <Int 2 ^Int 16
     andBool NONCE    <Int maxUInt256
 
+[execTransaction-checkSigs-exception]
+output: _ => _
+statusCode: _ => EVMC_REVERT
+proxy_storage:
+    store(store(store(M1, #hashedLocation({COMPILER}, {THRESHOLD}, .IntList),        THRESHOLD),
+                          #hashedLocation({COMPILER}, {NONCE}, .IntList),            NONCE),
+                          #hashedLocation({COMPILER}, {DOMAIN_SEPARATOR}, .IntList), DOMAIN_SEPARATOR)
+ => _
++requires:
+    ; enough signatures but exception
+    andBool SIGS_LEN >=Int THRESHOLD *Int 65
+    andBool #checkSignaturesException
+
+[execTransaction-checkSigs0]
+output: _ => _
+statusCode: _ => EVMC_REVERT
+proxy_storage:
+    store(store(store(M1, #hashedLocation({COMPILER}, {THRESHOLD}, .IntList),        THRESHOLD),
+                          #hashedLocation({COMPILER}, {NONCE}, .IntList),            NONCE),
+                          #hashedLocation({COMPILER}, {DOMAIN_SEPARATOR}, .IntList), DOMAIN_SEPARATOR)
++requires:
+    ; not enough signaures or invalid signatures
+    andBool ( SIGS_LEN <Int THRESHOLD *Int 65
+       orBool notBool #enoughValidSigs )
+
 [execTransaction-checkSigs1]
+proxy_storage:
+    store(store(store(M1, #hashedLocation({COMPILER}, {THRESHOLD}, .IntList),        THRESHOLD),
+                          #hashedLocation({COMPILER}, {NONCE}, .IntList),            NONCE => NONCE +Int 1),
+                          #hashedLocation({COMPILER}, {DOMAIN_SEPARATOR}, .IntList), DOMAIN_SEPARATOR)
```

Comment:
duplicating proxy_storage doesn't look good.  you can avoid it by introducing a parameter, say `NEW_NONCE`, and put the ensure, i.e., either `NEW_NONCE ==Int NONCE +Int 1` or `NEW_NONCE ==Int NONCE`.